### ISSUE TITLE
Finite field exception on read when running Lottery

### ISSUE THREAD

**Issue by psilospore** [ShapeOfMatter/MultiChor#69]
The Lottery example doesn't run due to an exception on reading the finite field
```
cabal run lottery client2
secret:
42
lottery: Failed to read "42" as a Fp
CallStack (from HasCallStack):
  error, called at examples/CLI.hs:49:38 in main:CLI
```

Something might just be wrong with the Read instance:

```haskell
ghci> import Lottery (Fp)
ghci> read @Fp "42"
Fp *** Exception: Prelude.read: no parse

-- But this works
read @(PrimeField 999983) "42"
42
```

I can get it to work with my own read instance rather than the derived one:
```haskell
instance Read Fp where
    readsPrec _ s = [(Fp (read s), "")]
```

Not sure why the newtype derived Read instance might be any different. I could push the version I have up if we don't want to investigate that much.

**Comment by ShapeOfMatter** [comment]
The derived `Read` of a `newtype` includes the newtype's constructor, so this works:
```haskell
ghci> read @Fp "Fp 42"
```
Let's keep this open; I think my suggested fix would be to edit the choreography to consume `Int`s and cast them internally. Or we could switch it to be a `type` instead of a `newtype` or we could write our own `instance Read` (and `Show`!).

I was tempted to say that this was of zero consequence, but actually things like this will be good to fix in time for artifact-review!

**Comment by ShapeOfMatter** [comment]
closed by #70
